data_test.go

```
@@ -44,8 +45,77 @@ func TestRunDataCommand(t *testing.T) {
 			checkResult(t, code, out.String(), errb.String(), tt.wantCode, tt.wantOut, tt.wantErr)
 		})
 	}
+
+	t.Run("json format emits datapoint objects", func(t *testing.T) {
+		var out, errb bytes.Buffer
+		code := runDataCommand([]string{"g"}, &FakeClient{FetchGoalWithDatapointsFunc: twoPoints}, "json", &out, &errb)
+		if code != 0 {
+			t.Fatalf("expected exit 0, got %d (stderr: %s)", code, errb.String())
+		}
+		var dps []Datapoint
+		if err := json.Unmarshal([]byte(out.String()), &dps); err != nil {
```

golangci-lint (2.12.2)

[error] S1030: should use out.Bytes() instead of []byte(out.String())

(staticcheck)